andela/mmdp_site
# MMDP-371 Allow guest access to list endpoints

**What does this PR do?**

Allows guests access to list endpoints.

**Description of Task to be completed?**

Ensure that guests do not see archived records.

**How should this be manually tested?**

- Fetch and checkout to the `ft-list-guest-access-MMDP-371` branch.
- On postman try accessing cms list endpoints. They should all allow unauthenticated access.

**Any background context you want to provide?**

Replaces `authentication` and `authorization` middleware with a new middleware called `authOptional` for all listing endpoints. This middleware adds `user` to the `req` object if the request is authenticated. This is what is used to differentiate admins from guests.

**What are the relevant Jira issues?**

[MMDP-371](https://viisaus.atlassian.net/browse/MMDP-371)

**Screenshots (if appropriate)**
##### Events list
Note the absence authorization header
![screenshot 2019-02-26 at 17 31 27](https://user-images.githubusercontent.com/8180548/53420347-aed88000-39ec-11e9-8010-b655395fdcae.png)

**Questions:**

N/A
